src/NServiceBus.Core/Pipeline/Outgoing/SerializeMessageConnector.cs

```diff
@@ -52,10 +53,16 @@ string SerializeEnclosedMessageTypes(Type messageType)
         {
             var metadata = messageMetadataRegistry.GetMessageMetadata(messageType);
 
-            var assemblyQualifiedNames = new HashSet<string>();
+            var assemblyQualifiedNames = new List<string>();
```

Comment:
Since we know the upper bound on the number of items, we should initialize the capacity to `metadata.MessageHierarchy.Length` to avoid multiple internal array allocations.

Reply:
yeah was thinking about this as well. Since I assumed that in most cases the hierarchy is quite small, the default size is a suitable default value but since the amount of duplicate message types should be super low, specifying a size seems to make sense. Will add that.